### PR TITLE
Set exitCode on failure to format stdin.

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: bin/format.dart [hunk 1]
```diff
@@ -149,11 +149,13 @@ void formatStdin(FormatterOptions options, List<int> selection) {
       return true;
     } on FormatterException catch (err) {
       stderr.writeln(err.message());
+      exitCode = 65;
```
Comment (munificent) [inline bin/format.dart]: Can you mention "from sysexits.h" here?

File: bin/format.dart [hunk 2]
```diff
@@ -149,11 +149,13 @@ void formatStdin(FormatterOptions options, List<int> selection) {
...
     } catch (err, stack) {
       stderr.writeln('''Hit a bug in the formatter when formatting stdin.
 Please report at: github.com/dart-lang/dart_style/issues
 $err
 $stack''');
+      exitCode = 66;
```
Comment (munificent) [inline bin/format.dart]: Let's do 70 (EX_SOFTWARE, internal software error) for this.

File: bin/format.dart (hunk 1 above)
Comment (cbracken) [inline bin/format.dart]: Done.

File: bin/format.dart (hunk 2 above)
Comment (cbracken) [inline bin/format.dart]: Done.

**Comment by munificent** [comment]
Looks nice! One more thing: can you add this to the CHANGELOG.md? Thanks!

**Comment by cbracken** [comment]
Done!

**Comment by munificent** [comment]
:+1:

**Pull request merged**